app/assets/javascripts/hideShowPassword.js

```diff
@@ -0,0 +1,453 @@
+(function (factory) {
```

Comment:
Do we really need all of this?  It seems that we could flip the type attribute on the password input in the same method js method that you're catching the checkbox event.  Is it not as simple as that?